gzread.c

```diff
@@ -24,7 +24,7 @@ local int gz_load(gz_statep state, unsigned char *buf, unsigned len,
             break;
         *have += (unsigned)ret;
     } while (*have < len);
-    if (ret < 0) {
+    if (ret < 0 && errno != EAGAIN && errno != EWOULDBLOCK) {
```

Comment:
This is the core of this PR, because all of the read-related wrappers here in gzread.c
end up calling this **gz_load** routine:  when
* **BOTH** state-fd (file descriptor) is non-blocking (fcntl(fd, SET_FL, fcntl(fd, GET_FL) | O_NONBLOCK))
* * **AND** there are no data available to read,
then the read(2) return value (**ret** above, assigned on line 22) will be -1, and errno will be either EWOULDBLOCK or EAGAIN (Resource not available).

The change here prevents that case from returning -1 from **gz_load()** here, which in turn
allows the calling routine\* to continue working with data from the file descriptor.

It gets a little tricky, because no bytes are returned, and 0 as the return value from **read(2)**
indicates EOF (End-Of-File).

N.B. tt may make sense to clear errno to a value of 0 here in gz_load(), just before line 22 above.

\* gz_avail() or gz_fetch() or gz_read(), with the latter being the ultimate caller of the first two.